addons/account/models/account_move.py

```diff
@@ -1120,6 +1130,13 @@ def _compute_suitable_journal_ids(self):
             domain = [('company_id', '=', company_id), ('type', '=', journal_type)]
             m.suitable_journal_ids = self.env['account.journal'].search(domain)
 
+    def _must_compute_name(self, current_move):
+        current_move.ensure_one()
+        # In the form view, we need to compute a default sequence so that the user can edit
+        # it. We only check the first move as an approximation (enough for new in form view)
+        highest_name = self[0]._get_last_sequence() if self else False
+        return not highest_name and current_move == self[0] and not current_move.posted_before and current_move.date
+
```

Comment:
Tried to keep as much as possible from the original code but yeah this is much clearer :)

Reply:
Oh I see, because we would only look at the first move. My proposition does lose this behavior. 
Then, you should make this one using self like proposed above and adapt the call below by only calling the method if move == self[0]

Reply:
Then this checked would be duplicated in the extended method in `enterprise/account_accountant/models/account_move.py`. The idea was to check everything inside this method to avoid this duplication of checks